graphql-dgs-client/src/test/kotlin/com/netflix/graphql/dgs/client/GraphqlSSESubscriptionGraphQLClientTest.kt

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2021 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.graphql.dgs.client
+
+import com.netflix.graphql.dgs.autoconfig.DgsAutoConfiguration
+import com.netflix.graphql.dgs.subscriptions.graphql.sse.DgsGraphqlSSEAutoConfig
+import com.netflix.graphql.dgs.subscriptions.sse.DgsSSEAutoConfig
+import com.netflix.graphql.dgs.subscriptions.websockets.DgsWebSocketAutoConfig
+import org.junit.jupiter.api.Assertions.assertThrows
+import org.junit.jupiter.api.Test
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.boot.test.web.server.LocalServerPort
+import org.springframework.web.reactive.function.client.WebClient
+import org.springframework.web.reactive.function.client.WebClientResponseException
+import reactor.test.StepVerifier
+
+@SpringBootTest(
+    classes = [DgsAutoConfiguration::class, DgsGraphqlSSEAutoConfig::class, TestApp::class],
+    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT
+)
+@EnableAutoConfiguration(exclude = [DgsSSEAutoConfig::class, DgsWebSocketAutoConfig::class])
```

Comment:
So the reason I had to exclude DgsSSEAutoConfig is because you can't have two rest controllers running on the same endpoint. And the reason I excluded WebSocketAutoConfig, is because the way I see these you a user wouldn't be mixing and matching subscription types. You either use websockets or graphql-sse or the old version of the protocol

Reply:
Thanks, makes sense!